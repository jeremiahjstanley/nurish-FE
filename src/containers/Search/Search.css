div .search-container {
  width: 45%;
  animation: grow-search 1s; }

@keyframes grow-search {
  0% {
    width: 25%; }
  100% {
    width: 45%; } }

.search-field {
  margin: 1rem auto;
  width: 95%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  height: 150px; }

.search-field .ui .button {
  background-color: #FFF;
  border: 1px solid rgba(34, 36, 38, 0.15);
  border-style: solid solid solid none; }
