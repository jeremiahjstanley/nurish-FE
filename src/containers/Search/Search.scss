
div {
	& .search-container {
			animation: grow-search 1s;
			min-width: 30rem;
			width: 45%;
			@keyframes grow-search {
	  
		  	0% {
		  		width: 25%;
		  	}
		  
		  	100% {
		    	width: 45%;
		  	}
			}
	}	
}

.search-field {
	display: flex;
	flex-direction: column;
	height: 150px;
	justify-content: space-evenly;
	margin: 1rem auto;
	width: 95%;
}

.search-field {
	& .ui {
		& .button {
			background-color: $orange;
			border-style: solid solid solid none;
			border: 1px solid rgba(34, 36, 38, .15);
		}
	}
}
