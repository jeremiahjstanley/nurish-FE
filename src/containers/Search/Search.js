import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Dropdown, Input, Button, Select } from 'semantic-ui-react';
import './Search.css';

const options = [
  { key: 'name', text: 'name', value: 'name' },
  { key: 'HPCP', text: 'HPCP', value: 'HPCP' },
  { key: 'allergen', text: 'allergen', value: 'allergen' },
]

class Search extends Component {
	constructor() {
		super()

		this.state = {
			input: '',
			criteria: ''
		}
	}

	handleSubmit = (event) => {
		event.preventDefault()
		console.log(this.state.input, this.state.criteria)
	}

	handleChange = (input) => {
		this.setState({input})
	}


	render() {
		return (
			<div className='form-container search-container'>
				<div className='link-container'>
					<NavLink to='/'>back</NavLink>
					<NavLink to='/filter'>filter</NavLink>
					<NavLink to='/browse'>browse</NavLink>
				</div>
				<form className='search-field' onSubmit={this.handleSubmit}>
				  <Input 
				  	type='text'
				  	placeholder='enter your criteria' 
				  	onChange={event => this.handleChange(event.target.value)}
				  	action >
				    <input value={this.state.value}/>
				    <Select compact options={options} defaultValue='name' value={this.state.criteria}/>
				    <Button type='submit'>Search</Button>
				  </Input>
				</form>
			</div>
		)
	}
}

export default withRouter(Search);