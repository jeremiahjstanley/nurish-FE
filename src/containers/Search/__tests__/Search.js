import React from 'react';
import { shallow } from 'enzyme';
import Search from '../Search';


describe('Search', () => {
    let wrapper
    let mockHandleChange
    let mockHandleSubmit
    let mockHandleInput

  beforeEach(() => {
    mockHandleChange = jest.fn()
    mockHandleSubmit = jest.fn()
    mockHandleInput = jest.fn()
    wrapper = shallow(<Search 
      handleChange={mockHandleChange}
      handleSubmit={mockHandleSubmit}
      handleInput={mockHandleInput}  
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})