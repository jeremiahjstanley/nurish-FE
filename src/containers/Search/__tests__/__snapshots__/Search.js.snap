// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    fetchResults={[Function]}
    handleChange={[Function]}
    handleInput={[Function]}
    handleSubmit={[Function]}
    history={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="link-container"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/"
          >
                    back
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/filter"
          >
                    filter
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/formulas"
          >
                    browse
          </NavLink>
</div>,
        <form
          className="search-field"
          onSubmit={[Function]}
>
          <Input
                    action="Search"
                    onChange={[Function]}
                    placeholder="enter your criteria"
                    type="text"
                    value={undefined}
          />
</form>,
      ],
      "className": "form-container search-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
>
              back
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/filter"
>
              filter
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/formulas"
>
              browse
</NavLink>,
          ],
          "className": "link-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "back",
              "to": "/",
            },
            "ref": null,
            "rendered": "back",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "filter",
              "to": "/filter",
            },
            "ref": null,
            "rendered": "filter",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "browse",
              "to": "/formulas",
            },
            "ref": null,
            "rendered": "browse",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Input
            action="Search"
            onChange={[Function]}
            placeholder="enter your criteria"
            type="text"
            value={undefined}
/>,
          "className": "search-field",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": "Search",
            "onChange": [Function],
            "placeholder": "enter your criteria",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="link-container"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/"
            >
                        back
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/filter"
            >
                        filter
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/formulas"
            >
                        browse
            </NavLink>
</div>,
          <form
            className="search-field"
            onSubmit={[Function]}
>
            <Input
                        action="Search"
                        onChange={[Function]}
                        placeholder="enter your criteria"
                        type="text"
                        value={undefined}
            />
</form>,
        ],
        "className": "form-container search-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
>
                back
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/filter"
>
                filter
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/formulas"
>
                browse
</NavLink>,
            ],
            "className": "link-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "back",
                "to": "/",
              },
              "ref": null,
              "rendered": "back",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "filter",
                "to": "/filter",
              },
              "ref": null,
              "rendered": "filter",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "browse",
                "to": "/formulas",
              },
              "ref": null,
              "rendered": "browse",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Input
              action="Search"
              onChange={[Function]}
              placeholder="enter your criteria"
              type="text"
              value={undefined}
/>,
            "className": "search-field",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "action": "Search",
              "onChange": [Function],
              "placeholder": "enter your criteria",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
